src/parsers/smt2/smt2parser.cpp

```diff
@@ -3204,13 +3208,23 @@ namespace smt2 {
             }
         }
     };
+
+    void free_parser(parser * p) { dealloc(p); }
 };
 
 bool parse_smt2_commands(cmd_context & ctx, std::istream & is, bool interactive, params_ref const & ps, char const * filename) {
     smt2::parser p(ctx, is, interactive, ps, filename);
     return p();
 }
 
+bool parse_smt2_commands_with_parser(class smt2::parser ** p, cmd_context & ctx, std::istream & is, bool interactive, params_ref const & ps, char const * filename) {
```

Comment:
smt2::parser *& ?

Reply:
I'll adopt that style. Without knowing better, I like using the '&' at the call site to signal that the parameter can be mutated (i.e. `parse_smt2_commands_with_parser(&p, ...)`).

Reply:
Done.